### DISCUSSION TITLE
Explicit valence when creating a MolBlock

### DISCUSSION THREAD

**Discussion by spparel** [rdkit/rdkit#7542] [General]
Hi,

I am facing the following issue. I need to register into a commercial cartridge solution structures that have been imported from SMILES into RDKit, got processed and are then re-exported as MolBlock. The basic process can be summarized as:

`mol = Chem.MolFromSmiles('COC(=O)c1ccc(CBr)c([N+](=O)[O-])c1')`
`resulting_mol=Chem.MolToMolBlock(mol, forceV3000=False)`
`print(resulting_mol) `

The result is a properly formatted MolBlock, but for one thing. The valence on the Nitrogen and on the negatively-charged oxygen of the nitro are explicitly indicated:

   -1.5000    2.5981    0.0000 N   0  0  0  0  0  **4**  0  0  0  0  0  0
   -3.0000    2.5981    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
   -0.7500    3.8971    0.0000 O   0  0  0  0  0  **1**  0  0  0  0  0  0

Somehow this confuses our registration system.

When I export the same structure to a molfile either from Biovia Draw or from Marvin Sketch, these valences are either 0 or not shown at all.

I'm not sure which approach is more correct (considering that N and O are part of the "organic set", I think). And is there a way to prevent RDKit from calculating and outputting these explicit valences ? I tried to set sanitize to False during the import, but the behavior is the same.

Thank you for your help

**Comment by marcostenta** [comment]
The desired behavior is to avoid printing out in the molblock the valence when it matches the accepted one for the atoms or in case of charged species, it isoelectric match.
now 

mol = Chem.MolFromSmiles('CC[NH3+]')
molblock = Chem.MolToMolBlock(mol, forceV3000=True)
print(molblock)


     RDKit          2D

  0  0  0  0  0  0  0  0  0  0999 V3000
M  V30 BEGIN CTAB
M  V30 COUNTS 3 2 0 0 0
M  V30 BEGIN ATOM
M  V30 1 C 0.000000 0.000000 0.000000 0
M  V30 2 C 1.299038 0.750000 0.000000 0
M  V30 3 N 2.598076 -0.000000 0.000000 0 CHG=1 VAL=**4**
M  V30 END ATOM
M  V30 BEGIN BOND
M  V30 1 1 1 2
M  V30 2 1 2 3
M  V30 END BOND
M  V30 END CTAB
M  END

it would be good to switch off this printing wiht a parameter of the MolToMolBlock method and have something like 

     RDKit          2D

  0  0  0  0  0  0  0  0  0  0999 V3000
M  V30 BEGIN CTAB
M  V30 COUNTS 3 2 0 0 0
M  V30 BEGIN ATOM
M  V30 1 C 0.000000 0.000000 0.000000 0
M  V30 2 C 1.299038 0.750000 0.000000 0
M  V30 3 N 2.598076 -0.000000 0.000000 0 CHG=1
M  V30 END ATOM
M  V30 BEGIN BOND
M  V30 1 1 1 2
M  V30 2 1 2 3
M  V30 END BOND
M  V30 END CTAB
M  END

That is read correctly by marvin js, and by RDKIT itself.

molblock_2 = """
     RDKit          2D
 
  0  0  0  0  0  0  0  0  0  0999 V3000
M  V30 BEGIN CTAB
M  V30 COUNTS 3 2 0 0 0
M  V30 BEGIN ATOM
M  V30 1 C 0.000000 0.000000 0.000000 0
M  V30 2 C 1.299038 0.750000 0.000000 0
M  V30 3 N 2.598076 -0.000000 0.000000 0 CHG=1
M  V30 END ATOM
M  V30 BEGIN BOND
M  V30 1 1 1 2
M  V30 2 1 2 3
M  V30 END BOND
M  V30 END CTAB
M  END
"""

rdmol = Chem.MolFromMolBlock(molblock_2)

**Comment by spparel** [comment]
@greglandrum @bp-kelley : May I kindly ask you to have a look at this thread, please ? This is a serious issue for us when registering structures from molfiles generated in RDKit

Thanks

**Comment by greglandrum** [comment]
I don't think it's a bug (since the output is correct), but I agree that it would be better if the RDKit didn't do this.
